### PR TITLE
🐝 (svg tester) add GitHub action to test all possible chart views

### PR DESCRIPTION
**Pull request by sophiamersmann**
- Adds a GitHub action to test all possible views of a subset of charts
- The action works exactly like the other action works
    - To review, it's probably easiest to look at the diff between `svg-compare.yml` and `svg-compare-all-views.yml`
- Both actions commit to the same branch in `owid-grapher-svgs`
    - Having all SVGs for one change on the same branch is convenient since I often pull from the repo to look at the SVG differences locally (if there are many differences, GitHub often fails too to load the page) 
    - Accidental content overwrites are impossible because both actions operate in different directories
- The verify step:
    - The verify script reads chart IDs to process from the `most-viewed-charts.txt` file in `owid-grapher-svgs`
    - The reference SVGs for all chart views are stored in `all-views/svg`
    - Reference SVGs and the `most-viewed-charts.txt` text file are added to the `owid-grapher-svgs` repo in https://github.com/owid/owid-grapher-svgs/pull/12

---

- I couldn't get `act` to run actions locally, so this GitHub action is not tested in any way
- That's why for now the action is configured to not run automatically. It can only be triggered by the click of a button on GitHub.
- Once it's properly tested, I'd configure it to run on push/pull.

### CODE REVIEW

**Comment by sophiamersmann** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3062?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #3043** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3043?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3062** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3062?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/3062?utm_source=stack-comment).

File: .github/workflows/svg-compare.yml [hunk 1]
```diff
@@ -35,13 +35,20 @@ jobs:
...
 
-            # Switch to or create a branch on that repo that matches the branch name we are on in the owid-grapher repo
+            # Create a branch on that repo that matches the branch name we are on in the owid-grapher repo
             # but only do this if we are not on master in owid-graphers (in this case we want to commit and push on master
             # in owid-grapher-svgs as well)
             - name: create owid-grapher-svgs local branch
               if: ${{ env.PUSH_BRANCH_NAME != 'master'}}
               working-directory: owid-grapher-svgs
-              run: git branch ${{ env.PUSH_BRANCH_NAME }} && git checkout ${{ env.PUSH_BRANCH_NAME }}
+              continue-on-error: true
```
Comment (sophiamersmann) [inline .github/workflows/svg-compare.yml:44]: If both actions commit to the same branch in `owid-grapher-svgs`, then creating a branch should be a no-op if the branch already exists. `git branch <b>` fails with exit code 1 if `<b>` exists, that's why I set `continue-on-error` to true. I wasn't sure whether the next line (the checkout) would run if the previous line failed. To be on the safe side, the branch is checked out in a separate step.

File: .github/workflows/svg-compare-all-views.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+name: SVG diff (all views)
+on: workflow_dispatch
```
Comment (sophiamersmann) [inline .github/workflows/svg-compare-all-views.yml:2]: For now, this action can only be triggered by button click on GitHub. Once it's properly tested (and I'm sure it actually gives useful feedback), I would set this to push/pull.

File: .github/workflows/svg-compare-all-views.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+              run: git checkout ${{ env.PUSH_BRANCH_NAME }}
+
+            # Run the verify tool overwriting any svgs. Stdout is piped to compare-result which will be a 0 byte file if everything works or contain failed grapher ids otherwise
+            - name: Generate SVGs and compare to reference
+              id: run-verify-graphs
+              continue-on-error: true
+              env:
+                  # using "ternary operator" from https://github.com/actions/runner/issues/409#issuecomment-752775072
+                  RM_ON_ERROR: ${{ env.PUSH_BRANCH_NAME == 'master' && '' || '--rm-on-error' }}
+              run: node --enable-source-maps itsJustJavascript/devTools/svgTester/verify-graphs.js -i owid-grapher-svgs/configs -o owid-grapher-svgs/all-views/svg -r owid-grapher-svgs/all-views/svg --ids-from-file owid-grapher-svgs/most-viewed-charts.txt --all-views $RM_ON_ERROR > compare-result
```
Comment (sophiamersmann) [inline .github/workflows/svg-compare-all-views.yml:60]: The workflow is identical to the current one. This is the only line of code that meaningfully changed.

**Review by danyx23 — APPROVED** [review]
Looks good! Let's merge this and see if there are any issues when running manually but I'm very excited to have this soon for all commits :)

**Pull request merged**